app/helpers/application_helper.rb

```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title
+    base_title = t "Ruby_onRails_Tutorial_Sample_App"
+    page_title.empty? ? base_title : page_title + " | " + base_title
```

Comment:
so sánh giúp a: nil?, empty?, blank?, present? và ở trường hợp này nên sử dụng cái nào ?

Reply:
nil? muốn check nil? của bất kỳ object nào và trả về boolean, nếu nil thì là true, ngược lại là false.
empty  để check xem object có empty không? nếu rỗng (empty) thì là true, ngược lại là false, không thể sử dụng với any object, không sử dung được với object bị nil, nó sử dụng được với array, hash, string.
blank? trả về true nếu object là giá trị là nil, false, empty, ngược lại false, dùng cho bất kỳ object nào.
present?  để check object có NOT nil, NOT false, hay là NOT emptyhay không? Giá trị trả về sẽ ngược với .blank?, và cũng dùng được cho bất kỳ Object nào.
Trường hợp này e nghĩ nên dùng  empty?